src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -0,0 +1,391 @@
+                }
+            }
+
+            private static int NetworkToHostOrder(int value)
+            {
+                if (!BitConverter.IsLittleEndian)
+                    return value;
+
+                return (((value >> 24) & 0xFF) | ((value >> 08) & 0xFF00) | ((value << 08) & 0xFF0000) | ((value << 24)));
+            }
+
+            private static unsafe int GetStringLength(sbyte* s, int maxLength)
+            {
+                int len;
+                for (len = 0; len < maxLength; len++, s++)
+                {
+                    if (*s == 0)
+                    {
+                        break;
+                    }
+                }
+                return len;
+            }
+
+            [MemberNotNull(nameof(_ids))]
+            [MemberNotNull(nameof(_byteOffsets))]
+            [MemberNotNull(nameof(_lengths))]
+            private unsafe void ReadIndex(string tzFilePath, Stream fs, int indexOffset, int dataOffset, Span<byte> buffer)
+            {
+                int indexSize = dataOffset - indexOffset;
+                int entrySize = Marshal.SizeOf(typeof(AndroidTzDataEntry));
+                int entryCount = indexSize / entrySize;
+
+                _byteOffsets = new int[entryCount];
+                _ids = new string[entryCount];
+                _lengths = new int[entryCount];
+
+                for (int i = 0; i < entryCount; ++i)
+                {
+                    AndroidTzDataEntry entry = ReadAt<AndroidTzDataEntry>(tzFilePath, fs, indexOffset + (entrySize*i), buffer);
+                    var p = (sbyte*)entry.id;
+
+                    _byteOffsets![i] = NetworkToHostOrder(entry.byteOffset) + dataOffset;
+                    _ids![i] = new string(p, 0, GetStringLength(p, 40), Encoding.ASCII);
+                    _lengths![i] = NetworkToHostOrder(entry.length);
+
+                    if (_lengths![i] < Marshal.SizeOf(typeof(AndroidTzDataHeader)))
+                    {
+                        throw new InvalidOperationException(SR.InvalidOperation_BadIndexLength);
+                    }
+                }
+            }
+
+            public string[] GetTimeZoneIds()
+            {
+                return _ids!;
+            }
+
+            public byte[] GetTimeZoneData(string id)
+            {
+                int i = Array.BinarySearch(_ids!, id, StringComparer.Ordinal);
+                if (i < 0)
+                {
+                    throw new InvalidOperationException(SR.Format(SR.TimeZoneNotFound_MissingData, id));
+                }
+
+                int offset = _byteOffsets![i];
+                int length = _lengths![i];
+                var buffer = new byte[length];
+                // Do we need to lock to prevent multithreading issues like the mono/mono implementation?
+                string tzFilePath = GetTimeZoneDirectory() + TimeZoneFileName;
```

Comment:
The ctor's local scope doesn't hit this, since the variable is defined in the ctor instead of being a field in the class.
Would it be preferred that a field is allocated for the class in lieu of having a second call to `GetTimeZoneDirectory`?

Reply:
> Would it be preferred that a field is allocated for the class in lieu of having a second call to GetTimeZoneDirectory?

yes

Reply:
Done.